openapi3/response.go

```diff
@@ -27,8 +27,21 @@ func (responses Responses) Default() *ResponseRef {
 	return responses["default"]
 }
 
+// Get returns a ResponseRef for the given status
+// If an exact match isn't initially found a patterned field is checked using
+// the first digit to determine the range (eg: 201 to 2XX)
 func (responses Responses) Get(status int) *ResponseRef {
-	return responses[strconv.FormatInt(int64(status), 10)]
+	st := strconv.FormatInt(int64(status), 10)
+	rref, ok := responses[strconv.FormatInt(int64(status), 10)]
+	if ok {
+		return rref
+	}
+	st = string(st[0]) + "XX"
```

Comment:
shouldn't this try `string(st[0]) + string(st[1]) + "X"` first?

Also, isn't this code missing `len(st)` checks?

Reply:
It looks like there are only the specific ranges allowed in the spec.
> Only the following range definitions are allowed: 1XX, 2XX, 3XX, 4XX, and 5XX.

Perhaps a switch on `st` gives us the same guarantees all things considered?

Reply:
yes please let's stick to the spec and `case` each branch. You may panic on unexpected data. Just mention that in the func's docs.

Reply:
Updated. Let me know if that works. I opted not to introduce a panic (even if the range is out of bounds) as the method didn't previously panic with invalid values -- perhaps this can change in a future release (although, I'd still probably prefer an error).